src/main/java/org/opentripplanner/routing/core/RoutingRequest.java

```diff
@@ -816,6 +834,10 @@ public RoutingRequest clone() {
             clone.bannedTrips = (HashMap<AgencyAndId, BannedStopSet>) bannedTrips.clone();
             clone.bannedStops = bannedStops.clone();
             clone.bannedStopsHard = bannedStopsHard.clone();
+            clone.whiteListedAgencies = (HashSet<String>) whiteListedAgencies.clone();
+            clone.whiteListedRoutes = whiteListedRoutes.clone();
+            clone.preferredAgencies = (HashSet<String>) preferredAgencies.clone();
+            clone.preferredRoutes = preferredRoutes.clone();
```

Comment:
Perhaps worth noting that in addition to adding whitelisting, this PR *fixes* an existing bug where `preferredAgencies` and `preferredRoutes` were not getting copied on `RoutingRequest.clone()`, which meant that those parameters could not be used on a per-request basis without breaking things for subsequent requests. I'm not sure if that bug is documented somewhere.

Reply:
That's true. It should probably be its own issue. I guess not having it in a separate pull request is fine, as long as this pull request is accepted.

Reply:
I think we can just leave this additional fix in this PR.